src/test/java/pages/AbstractPage.java

```diff
@@ -0,0 +1,23 @@
+package pages;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import java.time.Duration;
+
+public abstract class AbstractPage {
+
+    protected final WebDriver driver;
+
+    public AbstractPage(WebDriver driver) {
+        this.driver = driver;
+    }
+
+    protected WebElement findElement(By locator) {
```

Comment:
Good approach, but if you are using this method for finding element, why are you using elementToBeClickable check? may be it's better to check for visability? It does not need to be clickable in all cases

Reply:
Replaced it with clickElement method, will implement a findElement later if needed